### DISCUSSION TITLE
Trace for API changes

### DISCUSSION THREAD

**Discussion by seokho-son** [cloud-barista/cb-tumblebug#416] [Show and tell]
### 사용 방법 가이드
API 및 문서 변경 PR이 발생하면, 
댓글에 해당 PR을 참조 기록하고, 필요시 간단한 설명을 추가한다.

**Comment by seokho-son** [comment]
Update api documentation for vmgroup feature #415

**Comment by seokho-son** [comment]
Add feature for general TB object retrieve #417

**Comment by seokho-son** [comment]
Add tumblebug obj delete feature and update obj api #418

**Comment by seokho-son** [comment]
Add private IP in mcis status to enhance weavescope deployment #422

**Comment by seokho-son** [comment]
Enhance ssh username verification performance #423

**Comment by seokho-son** [comment]
Support massive vm deployment with vmgropup enhancement #434

**Comment by seokho-son** [comment]
Enhance error handing for provisioning and cmd phases #435

**Comment by seokho-son** [comment]
Add codes to handle SSHAccessPoint #436

**Reply by seokho-son** [reply]
해당 API 갱신이 이뤄진 PR. Fix agent installation bug and script update #437

**Comment by jihoon-seo** [comment]
Update REST API of LookupImage & LookupSpec #444

**Comment by seokho-son** [comment]
Provide SystemMessage to vm status object #475

**Comment by seokho-son** [comment]
Enhance swagger documentation and UI mechanism #485

**Comment by seokho-son** [comment]
Code cleaning with cb-store0.3.2 cb-sp0.3.7 cb-df0.3.3 integration #488

**Comment by seokho-son** [comment]
Change snakeCase to camelCase in MCIS #490

**Comment by seokho-son** [comment]
Change get method to post in spec image #496

**Comment by seokho-son** [comment]
Pre-Released v0.3.5
https://github.com/cloud-barista/cb-tumblebug/releases/tag/v0.3.5

**Comment by jihoon-seo** [comment]
Add `inspectResources` function #501

**Comment by jihoon-seo** [comment]
Add `inspectVMs` function #505

**Comment by seokho-son** [comment]
Add initial code for mcis and vm plan with location-based algo #511

**Comment by seokho-son** [comment]
Fix bugs and correct stains in api doc #551

**Comment by seokho-son** [comment]
Pre-Released v0.3.7
https://github.com/cloud-barista/cb-tumblebug/releases/tag/v0.3.7

**Comment by seokho-son** [comment]
Add MCIS Refine feature #572

**Comment by jihoon-seo** [comment]
Enhance fetchImages & fetchSpecs functions #567 

Enhance FilterSpecsByRange function #580

**Comment by seokho-son** [comment]
Enhance error handling (Update rest api for mcis force del option) #579

**Comment by jihoon-seo** [comment]
Reflect Spider's SG object updates #585

Add `InitConfig` function #594

Add API for `ListResourceId`, `ListMcisId`, `ListVmId` #598

**Comment by seokho-son** [comment]
Pre-Released v0.3.10
https://github.com/cloud-barista/cb-tumblebug/releases/tag/v0.3.10

**Comment by seokho-son** [comment]
Pre-Released v0.3.11
https://github.com/cloud-barista/cb-tumblebug/releases/tag/v0.3.11

**Comment by seokho-son** [comment]
remove underscore from variable names #684

**Comment by seokho-son** [comment]
Fix not aligned name of variables #699

**Comment by seokho-son** [comment]
Fix MCIS label input bug #712

**Comment by seokho-son** [comment]
Add list MCIS simple option #731

**Comment by jihoon-seo** [comment]
Fix some REST APIs methods from get to post #742

**Comment by jihoon-seo** [comment]
- #983
- #990
- #1015 
- #1016
- #1018 
- #1019

**Comment by jihoon-seo** [comment]
## [v0.5.0]
## ↓
- ...
- MCIR 관련
  - #1023
  - #1083
  - #1137
- Register/Inspect existing CSP resource & VM 관련
  - #1046
  - #1073
  - #1076
  - #1078
  - #1080
  - #1092
  - #1094
  - #1097
- cb-network (CLADnet; Cloud Adaptive Network) 관련
  - #1054
  - #1100
  - #1151
- MCIS 관련
  - #1069
  - #1116
- NS 관련
  - #1132

## ↑
## [v0.6.0]
## ↓

- NLB 관련
  - #1158
  - 참고: #1172
  - #1175
  - #1178
- TB VMGroup 관련
  - #1176
  - #1177
  - #1181
- DataDisk 관련
  - #1182
- ...
